py/desitarget/skies.py

```diff
@@ -0,0 +1,486 @@
+    sindecmin, sindecmax = np.sin(np.radians(decmin)), np.sin(np.radians(decmax))
+    spharea = (ramax-ramin)*np.degrees(sindecmax-sindecmin)
+
+    #ADM how many sky positions we need to generate to meet the minimum density requirements
+    nskies = int(spharea*nskymin)
+    #ADM how many sky positions to generate, given that we'll reject objects close to bad
+    #ADM sources. The factor of 10 was derived by trial-and-error...but this doesn't need
+    #ADM to be optimal as this algorithm should become more sophisticated
+    nchunk = nskies*10
+
+    #ADM arrays of GOOD sky positions to populate with coordinates
+    ragood, decgood = np.empty(nskies), np.empty(nskies)
+    #ADM lists of BAD sky positions to populate with coordinates
+    rabad, decbad = [], []
+
+    #ADM ngenerate will become zero when we generate enough GOOD sky positions
+    ngenerate = nskies
+
+    while ngenerate:
+
+        #ADM generate random points in RA and Dec (correctly distributed on the sphere)
+        log.info('Generated {} test positions...t = {:.1f}s'.format(nchunk,time()-start))
+        ra = np.random.uniform(ramin,ramax,nchunk)
+        dec = np.degrees(np.arcsin(1.-np.random.uniform(1-sindecmax,1-sindecmin,nchunk)))
+
+        #ADM set up the coordinate objects
+        cskies = SkyCoord(ra*u.degree, dec*u.degree)
+        cobjs = SkyCoord(objs["RA"]*u.degree, objs["DEC"]*u.degree)
+
+        #ADM split the objects up using a separation of 4*navoid arcseconds in order to
+        #ADM speed up the coordinate matching when we have some objects with large radii
+        sepsplit = 4*navoid
+        bigsepw = np.where(sep > sepsplit)[0]
+        smallsepw = np.where(sep <= sepsplit)[0]
+
+        #ADM set up a list of skies that don't match an object
+        goodskies = np.ones(len(cskies),dtype=bool)
+
+        #ADM guard against the case where there are no objects with small radii
+        if len(smallsepw) > 0:
+            #ADM match the small-separation objects and flag any skies that match such an object
+            log.info('Matching positions out to {:.1f} arcsec...t = {:.1f}s'
+                     .format(sepsplit,time()-start))
+            idskies, idobjs, d2d, _ = cobjs[smallsepw].search_around_sky(cskies,sepsplit*u.arcsec)
+            w = np.where(d2d.arcsec < sep[smallsepw[idobjs]])
+            #ADM remember to guard against the case with no bad positions
+            if len(w[0]) > 0:
+                goodskies[idskies[w]] = False
+
+        #ADM guard against the case where there are no objects with large radii
+        if len(bigsepw) > 0:
+            #ADM match the large-separation objects and flag any skies that match such an object
+            log.info('Matching additional positions out to {:.1f} arcsec...t = {:.1f}s'
+                     .format(maxrad,time()-start))
+            idskies, idobjs, d2d, _ = cobjs[bigsepw].search_around_sky(cskies,maxrad*u.arcsec)
+            import pdb ; pdb.set_trace()
```

Comment:
leftover `pdb.set_trace()`.  I hit this when testing on DR4 sweep-160p035-170p040.fits.  Please confirm that this case is working as expected and remove `pdb.set_trace()` or otherwise further debug why this was needed.

Reply:
Thanks, this highlighted a bug in my unit test coverage where none of the test objects flagged this clause in the code. I've removed this `pdb.set_trace()` and improved the unit test coverage.